subprojects/core/src/main/java/org/gradle/initialization/DefaultBuildRequestContext.java

```diff
@@ -20,12 +20,19 @@
 
 public class DefaultBuildRequestContext implements BuildRequestContext {
     private final BuildCancellationToken token;
+    private final BuildGateToken gate;
     private final BuildEventConsumer buildEventConsumer;
     private final BuildRequestMetaData metaData;
 
+    // TODO: Wire the gate token into other levels
```

Comment:
`BuildRequestContext` is used by all of the entry points (TAPI, launcher, integ tests) when starting a Gradle build.  I need to update each of those and decide if they should be creating a new build gate or reusing one from somewhere else.  Keeping the old constructor let me defer how to handle that. 

This means we _could_ push the build gate up into the TAPI (like the cancellation token) and allow a TAPI client to open/close the gate. I didn't intend on doing that.